daemons/controld/controld_execd.c

```diff
@@ -289,29 +291,31 @@ send_task_ok_ack(lrm_state_t *lrm_state, ha_msg_input_t *input,
 void
 lrm_op_callback(lrmd_event_data_t * op)
 {
-    const char *nodename = NULL;
-    lrm_state_t *lrm_state = NULL;
-
     CRM_CHECK(op != NULL, return);
+    switch (op->type) {
+        case lrmd_event_disconnect:
+            if (op->remote_nodename == NULL) {
+                /* If this is the local executor IPC connection, set the right
+                 * bits in the controller when the connection goes down.
+                 */
+                lrm_connection_destroy();
+            }
+            break;
 
-    /* determine the node name for this connection. */
-    nodename = op->remote_nodename ? op->remote_nodename : fsa_our_uname;
+        case lrmd_event_exec_complete:
+            {
+                const char *nodename = op->remote_nodename?
+                                       op->remote_nodename : fsa_our_uname;
```

Comment:
new ternary operator spacing convention?

Reply:
Not really, it didn't fit in one 80-char line

Reply:
In such cases I'd advocate either of these approaches (up to consideration):

```
            const char *nodename = op->remote_nodename ? op->remote_nodename
                                                       : fsa_our_uname;
```

where allowed with text width, otherwise

```
                const char *nodename = op->remote_nodename
                                           ? op->remote_nodename
                                           : fsa_our_uname;
```

What was the most eye catching though was the question mark immediately
after an expression.

Reply:
As much as I love the ternary, it does cause readability issues if it's not small. In this case, I decided to functionize that step, so it does fit on one line now, and I think the function is more readable in itself.